CONTRIBUTING.md

```diff
@@ -94,14 +94,15 @@ download and install anything:
 * On each documentation page, there should be an "Improve This Page" link at the very
   top.
 * Click on that link to open the respective source file (usually an `.rst` file in the
-  `doc` folder) on GitHub for editing online (you'll need a GitHub account).
+  `doc` folder or a `.py` file in the `examples` folder) on GitHub for editing online
+  (you'll need a GitHub account).
 * Make your desired changes.
 * When you're done, scroll to the bottom of the page.
 * Fill out the two fields under "Commit changes": the first is a short title describing
   your fixes; the second is a more detailed description of the changes. Try to be as
   detailed as possible and describe *why* you changed something.
-* Click on the "Commit changes" button to open a
-  [pull request (see below)](#pull-requests).
+* Choose "Create a new branch for this commit and start a pull request." and
+  click on the "Propose changes" button to open a [pull request (see below)](#pull-requests).
```

Comment:
Users should not commit directly to their master branch. Instead, they should commit to a new branch and submit a PR.

Reply:
Should we recommend pygmt python-contributors to make branches on the GenericMappingTools/pygmt repo rather than their own fork?

Reply:
Do you mean "python-maintainers"? I don't think "python-contributors" have write permission to the pygmt repo.

Reply:
They have triage permissions. Doesn't that allow branching?

Reply:
I don't think they're allowed.

https://docs.github.com/en/github/setting-up-and-managing-organizations-and-teams/repository-permission-levels-for-an-organization#permission-levels-for-repositories-owned-by-an-organization

Reply:
I cannot actually.

Reply:
Huh, should we bump up the python-contributors team to have "write" permissions then? It's a bit annoying having to add a new git remote to check out a new contributor's code locally, and Leo mentioned before at https://github.com/GenericMappingTools/pygmt/pull/415#issuecomment-614548276 that "write" privileges are overrated anyway (i.e. pretty standard for an open source repo).

Reply:
If we do, we should add instructions on adding it to your local git repo; for better or for worse GitHub seems to encourage making branches on your own fork.

Reply:
Better to have more discussions at https://github.com/GenericMappingTools/gmt/discussions/4732 before making changes.

Reply:
> If we do, we should add instructions on adding it to your local git repo; for better or for worse GitHub seems to encourage making branches on your own fork.

I think it's because it's easier for new Github users to do stuff on their own fork. The instructions can be a bit long though, so would be good if we can just link to a good tutorial instead.



> Better to have more discussions at [GenericMappingTools/gmt#4732](https://github.com/GenericMappingTools/gmt/discussions/4732) before making changes.

Ok, I think we can have a 'python'/'pygmt'-only restructure, and it's up to the rest of the GMT ecosystem to decide whether to make the change too. Will make a quick comment on that thread.

Reply:
> for better or for worse GitHub seems to encourage making branches on your own fork.

I have a fork in my own account, but the branches are created in the pygmt repo, not in my own fork. Is it because I have admin permission?

This is what I see:

![image](https://user-images.githubusercontent.com/3974108/111085397-a11bc780-84ed-11eb-9e06-84e5e3ab0a9e.png)